app/controllers/api/base_controller/generic.rb

```diff
@@ -94,6 +94,23 @@ def delete_resource(type, id = nil, _data = nil)
         delete_resource_action(klass, type, id)
       end
 
+      def copy_resource(type, id, data = {})
```

Comment:
@cynepco3hahue I would make the same recommendation as per @jntullo - if it is needed for another resource we can always review that later, and there may be a better way of sharing that between API controllers without giving it to *everyone* in the base class